src/charm.py

```diff
@@ -80,10 +82,52 @@ def _on_config_changed(self, _):
         else:
             self.unit.status = WaitingStatus("Waiting for Pebble in workload container")
 
+    def _create_service_account(self):
+        """Create a ServiceAccount for training-operator."""
+        self.unit.status = MaintenanceStatus("Creating ServiceAccount")
+        try:
+            K8sResources.create_service_account()
+        except ResourceHandlingFailed as e:
+            logging.error(
+                "Unable to create ServiceAccount for training-operator: %s", str(e)
+            )
+        else:
+            logging.info("Successfully created ServiceAccount for training-operator")
+            self.unit.status = ActiveStatus()
```

Comment:
I would omit this from this particular method. I think it's better off after the two calls to `_create_service_account` and `_create_cluster_role_binding` in your `_on_install` method.

My preference is to keep any status changes within the event handlers themselves, rather than in helper methods :)